### PR TITLE
Initial submission of generated pubsub protobuf classes

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
/cc @anthmgoogle

### CODE REVIEW

File: gcloud-java-pubsub/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Google Cloud Java Client for Pub/Sub
```
Comment (aozarov) [inline gcloud-java-pubsub/README.md]: can you change it based on a recent [README.md](https://github.com/GoogleCloudPlatform/gcloud-java/tree/master/gcloud-java-storage)

**Comment by aozarov** [comment]
I think we should discuses package names.

Until now, an API for a service XXX would live under `com.google.gcloud.xxx` and its SPI classes would live in the same maven module but using a shared package `com.google.gcloud.spi` (these included only few classes and were prefixed with the Service name).

We decided to prefix `gcloud-java` packages with com.google.**gcloud** in order to avoid any future
collisions with service specific classes or its low-level apiary/gRPC client classes.

I wonder if we should start nesting spi classes under their specific service (e.g. `com.google.gcloud.pubsub.spi`) and if the classes in this PR should use that package instead of `com.google.pubsub`.

As discussed earlier spi classes are publicly accessible (so can be used directly) however don't go through the same process (test-coverage, documentation,...), therefore we need some way to distinct between the two.

File: gcloud-java-pubsub/README.md (hunk 1 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/README.md]: Not all of the sections are relevant yet - there isn't an example application and there isn't documentation on testing to point to. Do you want me to include the remainder of the sections?

**Comment by garrettjonesgoogle** [comment]
The classes in this particular review are low-level proto classes; the package name in use here is determined by pubsub.proto and isn't under our control. The idiomatic wrapper class for pubsub will come in a later review, and I was intending on putting that under the spi namespace.

File: gcloud-java-pubsub/README.md (hunk 1 above)
Comment (aozarov) [inline gcloud-java-pubsub/README.md]: Yes, and you can keep the left as comments with "//todo", so it will be easy to fix it later.

**Comment by aozarov** [comment]
Oh, I see. Sounds good.

Differently from gax, this is going to stay in this repository.
Any reason to check in the proto-generated classes instead of generating it at build time?
Looks like maven [supports](http://sergei-ivanov.github.io/maven-protoc-plugin/) it, right?

**Comment by garrettjonesgoogle** [comment]
Actually, I don't know if it is finalized that the generated pubsub classes will live here. We'd like to check these in now for that reason; trying to get maven to compile these might just be throwaway work. 

PS I also updated the README.md file per your request.

**Comment by aozarov** [comment]
Ok, I assumed using the Maven Protoc Plugin is [trivial](http://sergei-ivanov.github.io/maven-protoc-plugin/usage.html) but as you said we can change it later.

Yes, I think it would be great if there is a separate maven artifact for each service protobuf generated classes. Is that the plan?

**Pull request merged**